share/blob.go

```diff
@@ -124,6 +127,12 @@ func (b *Blob) Compare(other *Blob) int {
 	return b.namespace.Compare(other.namespace)
 }
 
+// IsEmpty returns true if the blob is empty. This is an invalid
```

Comment:
I would expect `blob.IsEmpty` to check if the data inside Blob was empty. Since this checks if the namespace is Empty, perhaps:

```suggestion
// IsEmpty returns true if the blob's namespace is empty. This is an invalid
```

Reply:
It checks the namespace as proxy for the entire blob. By construction you can't have a blob with empty data and a non-empty namespace. I could have checked that all the contents of the blob had their "null" value but only one should suffice

Reply:
I propose then doing `blob.Namespace().IsEmpty()` instead of proxying to avoid confusion.